### PR TITLE
Add Github actions for DMG and EXE packages

### LINKED ISSUE #29: Provide DMG and EXE packages
We should provide DMG and EXE packages for easier usage of Impose.

Furthermore, on MacOS with Apple Silicon installing Impose via `pip` will likely fail due to an issue with `h5py`, so a DMG package is currently the only option to run Impose on this platform.

### PR DESCRIPTION
**Pull request by raimund-schluessler**
This enhances the Github check action to build installer packages for MacOSX and Win. The packages are uploaded as artifacts to Github and can be downloaded from the Github Actions page.

Closes #29.

What is still missing is an Action to create packages and a release when a new tag is pushed. But I would propose to add this in a follow-up PR, since this one is already huge.

### CODE REVIEW

**Comment by raimund-schluessler** [comment]
For whatever reason actions don't run at all at the moment. 😕

**Comment by paulmueller** [comment]
I think the build pipeline should be fine like this. There have been a few updates (pyinstaller, scipy).

**Comment by raimund-schluessler** [comment]
The windows installer works fine.
The Mac DMG says "“[Impose.app](http://impose.app/)” is damaged and can’t be opened. You should move it to the Bin."

**Comment by raimund-schluessler** [comment]
> The windows installer works fine.

I have to correct myself. Installing Impose uninstalls BMicro and vice-versa.

File: build-recipes/win_Impose.iss_dummy [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+#define MyAppURL "https://impose.readthedocs.io"
+#define MyAppPlatform "win32"
+#define MyAppExeName "Impose.exe"
+#define MyAppDir = SourcePath + "dist\Impose\"
+
+[Setup]
+; NOTE: The value of AppId uniquely identifies this application.
+; Do not use the same AppId value in installers for other applications.
+; (To generate a new GUID, click Tools | Generate GUID inside the IDE.)
+AppId={{9700CB56-B733-14CD-ABBC-CAC320CE7C89}
```
Comment (raimund-schluessler) [inline build-recipes/win_Impose.iss_dummy:17]: We need to generate a new app id for Impose here. Otherwise it collides with the one for BMicro.

**Comment by raimund-schluessler** [comment]
> > The windows installer works fine.
> 
> I have to correct myself. Installing Impose uninstalls BMicro and vice-versa.

Windows installer works fine now. Lives happily together with BMicro. 👍

**Comment by raimund-schluessler** [comment]
But getting the tag in the _version.py script does not work. Not sure if only related to this PR or not.
The installer script generates a file named `Impose_2022.05.05.post08_win_64bit_setup.exe` whereas it should read `Impose_0.2.2.post08_win_64bit_setup.exe` (not the current date, but the tag/version 0.2.2).

**Comment by raimund-schluessler** [comment]
> Is there an option for GHA to also pull/fetch the tags? The tags are important.

I think that's not the issue. It also fails locally now for me.

Debugging `_version.py` shows that the call to git does not return something meaningful, but gives
```
b"fatal: unsafe repository ('D:/Data/Biotec/Software/00_Programs/BMicro' is owned by someone else)\nTo add an exception for this directory, call:\n\n\tgit config --global --add safe.directory D:/Data/Biotec/Software/00_Programs/BMicro\n"
```

**Comment by raimund-schluessler** [comment]
> The Mac DMG says "“[Impose.app](http://impose.app/)” is damaged and can’t be opened. You should move it to the Bin."

In principle the DMG seems to be fine. But when you download it via browser, Mac somehow flags it as unsafe and acknowledges running the DMG with this obscure, if not downright wrong, error message.

Downloading it on a windows machine and then copying the DMG file via a USB stick works flawlessly. Obviously, USB sticks have never been used to transfer malware and can be considered safe without any problem 🙄

**Pull request merged**